File: telegram_bot/engineerbot/ticket/ticket.go
package ticket

import (
	"encoding/json"
	"engineerbot/bot"
	"io"
	"log"
	"net/http"
	"strconv"
	"time"

	tgbotapi "github.com/go-telegram-bot-api/telegram-bot-api"
)

type Ticket struct {
	ID                    int64  `json:"id"`
	Subject               string `json:"subject"`
	CreatedAt             time.Time `json:"created_at"`
	Client                string `json:"client"`
	EquipmentName         string `json:"equipment_name"`
	SerialNumber          string `json:"serial_number"`
	CriticalityName       string `json:"criticality_name"`
	Hours                 string `json:"hours"`
	Description           string `json:"description"`
	Engineer              string `json:"engineer"`
	AdditionalInformation string `json:"additional_information"`
	Address               string `json:"address"`
	ChatID                int64  `json:"chat_id"`
}


type Handler struct {
	bot *bot.Bot
}

func NewHandler(bot *bot.Bot) *Handler {
	return &Handler{bot}
}

func (h *Handler) HandleNewTicket(w http.ResponseWriter, r *http.Request) {

	// Set CORS headers
    w.Header().Set("Access-Control-Allow-Origin", "http://localhost:3000")
	w.Header().Set("Access-Control-Allow-Methods", "POST")
	w.Header().Set("Access-Control-Allow-Headers", "Content-Type")

	// If this is a preflight request, we only need to return the headers above.
	if r.Method == http.MethodOptions {
		return
	}

	body, err := io.ReadAll(r.Body)
	if err != nil {
		http.Error(w, "Failed to read request body", http.StatusBadRequest)
		return
	}
	defer r.Body.Close()

	var ticket Ticket
	err = json.Unmarshal(body, &ticket)
	if err != nil {
		http.Error(w, "Failed to parse JSON", http.StatusBadRequest)
		return
	}
	ticketIDStr := strconv.FormatInt(ticket.ID, 10)
	messageText := "New ticket:\n" +
		"ID: " + ticketIDStr + "\n" +
		"Subject: " + ticket.Subject + "\n" +
		"Created At: " + ticket.CreatedAt.String() + "\n" +
		"Client: " + ticket.Client + "\n" +
		"Equipment Name: " + ticket.EquipmentName + "\n" +
		"Serial Number: " + ticket.SerialNumber + "\n" +
		"Criticality Name: " + ticket.CriticalityName + "\n" +
		"Hours: " + ticket.Hours + "\n" +
		"Description: " + ticket.Description + "\n" +
		"Address: " + ticket.Address + "\n" +
		"Additional Information: " + ticket.AdditionalInformation + "\n"

	msg := tgbotapi.NewMessage(ticket.ChatID, messageText)
	msg.ReplyMarkup = tgbotapi.NewInlineKeyboardMarkup(
		tgbotapi.NewInlineKeyboardRow(
			tgbotapi.NewInlineKeyboardButtonData("Took the ticket", "TookTicket:"+ticketIDStr),
			tgbotapi.NewInlineKeyboardButtonData("Finished", "Finished:"+ticketIDStr),
		),
	)

	if _, err := h.bot.BotAPI.Send(msg); err != nil {
		log.Panic(err)
	}
}

File: telegram_bot/engineerbotv2/ticket.py
import json
from flask import jsonify, make_response
from telegram import InlineKeyboardMarkup, InlineKeyboardButton

class Ticket:
    def __init__(self, data):
        self.data = data

class Handler:
    
    def __init__(self, bot):
        self.bot = bot

    def handle_new_ticket(self, request):
        if request.method == 'OPTIONS':
            # This is a preflight request. Reply successfully:
            response = make_response()
            response.headers.add("Access-Control-Allow-Origin", "http://localhost:3000")
            response.headers.add('Access-Control-Allow-Headers', "Content-Type")
            response.headers.add('Access-Control-Allow-Methods', "POST")
            return response

        ticket = Ticket(request.json)
        ticket_id_str = str(ticket.data['id'])

        message_text = f"""New ticket:
ID: {ticket_id_str}
Subject: {ticket.data['subject']}
Created At: {ticket.data['created_at']}
Client: {ticket.data['client']}
Equipment Name: {ticket.data['equipment_name']}
Serial Number: {ticket.data['serial_number']}
Criticality Name: {ticket.data['criticality_name']}
Hours: {ticket.data['hours']}
Description: {ticket.data['description']}
Address: {ticket.data['address']}
Additional Information: {ticket.data['additional_information']}"""

        keyboard = [
            [
                InlineKeyboardButton("Took the ticket", callback_data='TookTicket:'+ticket_id_str),
                InlineKeyboardButton("Finished", callback_data='Finished:'+ticket_id_str),
            ]
        ]
        reply_markup = InlineKeyboardMarkup(keyboard)
        self.bot.send_message(chat_id=ticket.data['chat_id'], text=message_text, reply_markup=reply_markup)